test/remote/db_tests.js

```diff
@@ -621,6 +623,49 @@ test(
 )
 
 test(
+  'unblock code',
+  function (t) {
+    var unblockCode;
+    return dbConn.then(function(db) {
+      return db.createUnblockCode(ACCOUNT.uid)
+      .then(function(_unblockCode) {
+        t.ok(_unblockCode);
+        unblockCode = _unblockCode;
+
+        return db.consumeUnblockCode(ACCOUNT.uid, 'NOTREAL')
+      })
+      .then(
+        function () {
+          t.fail('consumeUnblockCode() with an invalid unblock code should not succeed')
+        },
+        function (err) {
+          t.equal(err.errno, 127, 'consumeUnblockCode() fails with the correct error code')
+          var msg = 'Error: Invalid unblock code';
+          t.equal(msg, '' + err, 'consumeUnblockCode() fails with the correct message')
+        }
+      )
+      .then(function() {
+        return db.consumeUnblockCode(ACCOUNT.uid, unblockCode);
+      })
```

Comment:
I'd insert a rejection handler at this point, otherwise a rejection from the valid call will fall down to line 659 where you test the rejection that you're expecting. Or have I read the logic wrong?